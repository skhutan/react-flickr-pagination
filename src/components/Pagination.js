import React from 'react';
import { Flex, Button, Text } from 'rebass';

const Pagination = ({
  currentPage,
  lastPage,
  handlePreviousPage,
  handleNextPage
}) => (
  <Flex alignItems="center" justifyContent="space-evenly" marginTop="1rem">
    <Button onClick={handlePreviousPage}>Previous Page</Button>
    <Text fontSize={2} fontWeight="bold">
      {currentPage}/{lastPage}
    </Text>
    <Button onClick={handleNextPage}>Next Page</Button>
  </Flex>
);

export default React.memo(Pagination);
