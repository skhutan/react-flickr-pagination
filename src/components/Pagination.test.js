import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pagination from './Pagination';

describe('Pagination', () => {
  it('renders correctly as a snapshot', () => {
    const wrapper = mount(
      <Pagination
        currentPage={1}
        lastPage={10}
        handleNextPage={jest.fn()}
        handlePreviousPage={jest.fn()}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
