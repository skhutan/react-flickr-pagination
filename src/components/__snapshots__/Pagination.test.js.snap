// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly as a snapshot 1`] = `
<Memo(Pagination)
  currentPage={1}
  handleNextPage={[MockFunction]}
  handlePreviousPage={[MockFunction]}
  lastPage={10}
>
  <Styled(div)
    alignItems="center"
    justifyContent="space-evenly"
    marginTop="1rem"
  >
    <div
      className="css-9yaqjg"
    >
      <ForwardRef
        onClick={[MockFunction]}
      >
        <Styled(div)
          __css={
            Object {
              "appearance": "none",
              "bg": "primary",
              "border": 0,
              "borderRadius": 4,
              "color": "white",
              "display": "inline-block",
              "fontSize": "inherit",
              "lineHeight": "inherit",
              "px": 3,
              "py": 2,
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
          as="button"
          onClick={[MockFunction]}
          tx="buttons"
          variant="primary"
        >
          <button
            className="css-vhv7r1"
            onClick={[MockFunction]}
          >
            Previous Page
          </button>
        </Styled(div)>
      </ForwardRef>
      <ForwardRef
        fontSize={2}
        fontWeight="bold"
      >
        <Styled(div)
          fontSize={2}
          fontWeight="bold"
          tx="text"
        >
          <div
            className="css-fm3b53"
          >
            1
            /
            10
          </div>
        </Styled(div)>
      </ForwardRef>
      <ForwardRef
        onClick={[MockFunction]}
      >
        <Styled(div)
          __css={
            Object {
              "appearance": "none",
              "bg": "primary",
              "border": 0,
              "borderRadius": 4,
              "color": "white",
              "display": "inline-block",
              "fontSize": "inherit",
              "lineHeight": "inherit",
              "px": 3,
              "py": 2,
              "textAlign": "center",
              "textDecoration": "none",
            }
          }
          as="button"
          onClick={[MockFunction]}
          tx="buttons"
          variant="primary"
        >
          <button
            className="css-vhv7r1"
            onClick={[MockFunction]}
          >
            Next Page
          </button>
        </Styled(div)>
      </ForwardRef>
    </div>
  </Styled(div)>
</Memo(Pagination)>
`;
